### PR TITLE
alloc: fix memory heap initialization

### PR DESCRIPTION
**Pull request by libinyang**
This patch fixes the bug of runtime & buffer memory heap initialization and
sets the correct base for each map.
This is a generic fix and It was verified to fix the issues below:
https://github.com/thesofproject/linux/issues/253
https://github.com/thesofproject/sof/issues/539 duplicate of linux #253

Signed-off-by: Libin Yang <libin.yang@intel.com>

### CODE REVIEW

**Comment by libinyang** [comment]
@lgirdwood 
@tlauda 

Hi all,

This is a patch to fix the bug of memory initialization in FW.

**Comment by wenqingfu** [comment]
Is there a corresponding issue number of the "bug"? @libinyang

**Comment by libinyang** [comment]
@wenqingfu 
@lgirdwood 
@mmaka1 
@tlauda 

This patch is made for the bug #539. However, as the memory issue should be very general, I believe this patch may fix some other potential unstable issues because of the memory allocation.

This bug is the member base of memmap.runtime[] map and memmap.buffer[] map is set wrong.

File: src/lib/alloc.c [hunk 1]
```diff
@@ -613,9 +613,7 @@ void init_heap(struct sof *sof)
 	struct mm_heap *heap;
 	struct block_map *next_map;
 	struct block_map *current_map;
-	int i;
-	int j;
-	int k;
+	int i, j;
```
Comment (lgirdwood) [inline src/lib/alloc.c]: please follow coding style of one variable per line.

File: src/lib/alloc.c [hunk 2]
```diff
@@ -626,21 +624,19 @@ void init_heap(struct sof *sof)
 	/* initialise buffer map */
 	for (i = 0; i < PLATFORM_HEAP_BUFFER; i++) {
 		heap = &memmap.buffer[i];
-
-		for (j = 0; j < heap->blocks; j++) {
-
```
Comment (lgirdwood) [inline src/lib/alloc.c]: keep the new line

File: src/lib/alloc.c [hunk 3]
```diff
@@ -649,21 +645,19 @@ void init_heap(struct sof *sof)
 	/* initialise runtime map */
 	for (i = 0; i < PLATFORM_HEAP_RUNTIME; i++) {
 		heap = &memmap.runtime[i];
-
-		for (j = 0; j < heap->blocks; j++) {
-
```
Comment (lgirdwood) [inline src/lib/alloc.c]: keep new line

File: src/lib/alloc.c (hunk 1 above)
Comment (libinyang) [reply src/lib/alloc.c]: > please follow coding style of one variable per line.

Thanks for point it out. I didn't realize we have such coding style in sof project.

File: src/lib/alloc.c (hunk 3 above)
Comment (libinyang) [reply src/lib/alloc.c]: OK for this and upper review comments.

**Comment by plbossart** [comment]
this change makes issue https://github.com/thesofproject/linux/issues/253 go away, can we please prioritize reviews and integration.

**Pull request merged**